doc/user/interop.md

```diff
@@ -115,19 +116,24 @@ If the name starts with `@`, write it as an instance variable:
 
 `WRITE(object, "@name", value)` → `object.instance_variable_set("name", value)`
 
-Otherwise, if there isn't a method defined on the object with the same name as
-the name, and there is a method defined on the object called `[]=`, call `[]=`
-with the name and value as the two arguments:
+Otherwise, if there is a method defined on the object with the same name as
+the name appended with `=`, and there isn't a method defined on the object
+called `[]=`, perform a method call using the name appended with `=` as the
+called method name, and the value as the argument:
 
-`WRITE(object, name, value)` → `object[name] = value` unless
-`object.responds_to?(name)`
+`WRITE(object, name, value)` → `object.name = value` if
+`object.responds_to?(name + "=")` and unless `object.responds_to?("[]=")`
 
-Otherwise, perform a method call using the name appended with `=` as the called
-method name, and the value as the argument:
+Otherwise, if there isn't a method defined on the object with the same name as
+the name appended with `=`, and there is a method defined on the object called
+`[]=`, call `[]=` with the name and value as the two arguments:
```

Comment:
I'm making it always prefer `name=`, even if there is a `[]=`, and to make that choice deterministic rather than depending on prior specialisation order.